app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb

```diff
@@ -7,6 +7,73 @@ def available_vms
     availability_zone.vms
   end
 
+  def self.params_for_create()
+    {
+      :fields => [
+        {
+          :component => 'switch',
+          :name      => 'shareable',
+          :id        => 'shareable',
+          :label     => _('Shareable'),
+          :onText    => _('Yes'),
+          :offText   => _('No')
+        },
+        {
+          :component    => 'select',
+          :name         => 'affinity_policy',
+          :id           => 'affinity_policy',
+          :label        => _('Affinity Policy'),
+          :initialValue => 'off',
+          :options      => [
+            {
+              :label => 'Off',
+              :value => 'off'
+            },
+            {
+              :label => 'Affinity',
+              :value => 'affinity'
+            },
+            {
+              :label => 'Anti-affinity',
+              :value => 'anti-affinity'
+            }
+          ]
+        },
+        {
+          :component => 'select',
+          :name      => 'volume_type',
+          :id        => 'volume_type',
+          :label     => _('Cloud Volume Type'),
+          :options   => self.class.CLOUD_VOLUME_TYPES, # TODO: the types are in the angular controller, need to extract them
+          :condition => {
+            :when => 'affinity_policy',
+            :is   => 'off'
+          },
+          :validate  => [{:type => 'required'}]
+        },
+        {
+          :component => 'select',
+          :name      => 'affinity_volume_id',
```

Comment:
Same here, I don't see this anywhere in the DB...

Reply:
All of the affinity controls/info will be a new feature for us.

Reply:
Okay, I'm going to omit this from the schema and it should be re-added when the feature is implemented...

Reply:
I'm OK with that plan - remove it and I'll add it back later when I implement affinity controls.

I don't fully understand why each field needs to be associated with a DB column, though. Would it not be possible to pass some general parameter to our IBM Cloud API without persisting anything to the DB?

Reply:
@agrare this question is for you I guess

Reply:
I think we should try to keep the params similar to the db columns if they exist (e.g. shareable vs multi_attachment above) but if it doesn't exist yet there isn't a requirement that every param match a db column for operations.

Reply:
We just couldn't reuse these same params to then implement an "edit" form